test/AST-Quake/vector-ctor-sized.cpp

```diff
@@ -0,0 +1,46 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+// Test for std::vector initializer_list constructor support
+
+// RUN: cudaq-quake %s | cudaq-opt --canonicalize | FileCheck %s
+
+#include "cudaq.h"
+
+__qpu__ void test() {
+  cudaq::qubit q;
+  std::vector<double> angle(2);   
+  angle[0] = M_PI_2;
+  angle[1] = M_PI_4;
```

Comment:
Do we still want to add plain old C array support? It was on the list once upon a time, but it doesn't look finished to me.

Reply:
I don't think at this point we should be allowing allocation via `malloc` within kernel code. I do think that we should support host code passing any array data via `std::span`.

Reply:
Right. I was thinking specifically of something like
```
double array[4];
array[0] = 22.2;
```

Reply:
If we support that, would we need also tests for these additional variations?
```
double array[4] = {1,2,3,4};
double array[4] = {1,2};      // implicit initialization of last 2 elements to 0
double array[]  = {1,2,3,4};  // no length specified but should probably still resolve to the same as the first one.
```

Reply:
This I do think we should support. But it should be separate from this PR

Reply:
Yes and yes. We do not support local arrays at the moment, but when we do in a future PR ...